compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -271,7 +271,7 @@ public BLangPackage genBIR(BLangPackage astPkg) {
         this.birOptimizer.optimizePackage(birPkg);
         astPkg.symbol.birPackageFile = new BIRPackageFile(new BIRBinaryWriter(birPkg).serialize());
 
-        if (astPkg.hasTestablePackage()) {
+        if (astPkg.hasTestablePackage() && !astPkg.moduleContextDataHolder.skipTests()) {
```

Comment:
Should we first check astPkg.moduleContextDataHolder.skipTests()?

Reply:
Addressed in 12a286f